src/Orleans.Reminders/Hosting/SiloBuilderReminderExtensions.cs

```diff
@@ -0,0 +1,39 @@
+using Microsoft.Extensions.DependencyInjection;
+using Orleans.Runtime;
+using Orleans.Configuration.Internal;
+using System.Linq;
+using Orleans.Runtime.ReminderService;
+using Orleans.Providers;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Orleans.Timers;
+using Orleans.Reminders;
+
+namespace Orleans.Hosting;
+
+public static class SiloBuilderReminderExtensions
+{
+    /// <summary>
+    /// Adds support for reminders to this silo.
+    /// </summary>
+    /// <param name="builder">The builder.</param>
+    /// <returns>The silo builder.</returns>
+    public static ISiloBuilder AddReminders(this ISiloBuilder builder) => builder.ConfigureServices(services => AddReminders(services));
+
+    /// <summary>
+    /// Add support for reminders to this client.
+    /// </summary>
+    /// <param name="services">The services.</param>
+    private static void AddReminders(this IServiceCollection services)
+    {
+        if (services.Any(service => service.ServiceType.Equals(typeof(ReminderProviderRuntime))))
+        {
+            return;
+        }
+        services.TryAddSingleton<IReminderRegistry, ReminderRegistry>();
+        services.AddTransient<IConfigurationValidator, ReminderOptionsValidator>();
+        services.AddSingleton<ReminderProviderRuntime>();
+        services.AddFromExisting<IReminderProviderRuntime, ReminderProviderRuntime>();
+        services.AddSingletonNamedService<IReminderService,LocalReminderService>(RemindersConstants.LocalReminderService);
```

Comment:
For future work, I think we can use reminder service engines that use different loading and management semantics. This is in preparation of that.

Reply:
removed this for now